codegen/scala-gen/src/main/scala/com/akkaserverless/codegen/scalasdk/SourceGenerator.scala

```diff
@@ -19,24 +19,80 @@ package com.akkaserverless.codegen.scalasdk
 import com.lightbend.akkasls.codegen.ModelBuilder
 import protocgen.CodeGenResponse
 
+import scala.annotation.tailrec
+
 object SourceGenerator {
 
+  val managedComment = """/* This code is managed by Akka Serverless tooling.
+                         | * It will be re-generated to reflect any changes to your protobuf definitions.
+                         | * DO NOT EDIT
+                         | */""".stripMargin
+
+  val unmanagedComment = """|/* This code was generated by Akka Serverless tooling.
+                            | * As long as this file exists it will not be re-generated.
+                            | * You are free to make changes to this file.
+                            | */""".stripMargin
+
   /**
    * Generate the 'managed' code for this model: code that will be regenerated regularly in the 'compile' configuratio
    */
   def generateManaged(model: ModelBuilder.Model): Seq[File] =
     Seq(File("foo/bar/AbstractBaz.scala", "package foo.bar\n\nabstract class AbstractBaz"))
+      .map(_.prependComment(managedComment))
 
   /**
    * Generate the 'managed' code for this model: code that will be regenerated regularly in the 'compile' configuratio
    */
   def generateManagedTest(model: ModelBuilder.Model): Seq[File] =
     Seq(File("foo/bar/BazSpec.scala", "package foo.bar\n\nclass BazSpec { new Baz() }"))
+      .map(_.prependComment(managedComment))
 
   /**
    * Generate the 'unmanaged' code for this model: code that is generated once on demand and then maintained by the
    * user.
    */
   def generateUnmanaged(model: ModelBuilder.Model): Seq[File] =
-    Seq(File("foo/bar/Baz.scala", "package foo.bar\n\nclass Baz extends AbstractBaz"))
+    Seq(File("foo/bar/Baz.scala", "package foo.bar\n\nclass Baz extends AbstractBaz"), generateMain(model))
+      .map(_.prependComment(unmanagedComment))
+
+  def generateMain(model: ModelBuilder.Model): File = {
+    val mainPackage = mainPackageName(model.services.keys ++ model.entities.keys)
+    File(
+      (mainPackage :+ "Main.scala").mkString("/"),
+      s"""
+         |package ${mainPackage.mkString(".")}
+         |
+         |object Main extends App {
+         |  println("Hello, world!")
+         |}
+         |""".stripMargin)
+  }
+
+  def mainPackageName(classNames: Iterable[String]): List[String] = {
+    val packages = classNames
+      .map(
+        _.replaceFirst("\\.[^.]*$", "")
+          .split("\\.")
+          .toList)
+      .toSet
+    if (packages.isEmpty) throw new IllegalArgumentException("Nothing to generate!")
+    longestCommonPrefix(packages.head, packages.tail)
+  }
+
+  @tailrec
+  def longestCommonPrefix(
```

Comment:
this has moved to SourceGeneratorUtils in https://github.com/lightbend/akkaserverless-java-sdk/pull/422